control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1396,6 +1396,22 @@ func (r *reconciler) ensureIngressControllersRemoved(ctx context.Context) (bool,
 	if len(errs) > 0 {
 		return false, fmt.Errorf("failed to delete ingress controllers: %w", errors.NewAggregate(errs))
 	}
+
+	routerPods := &corev1.PodList{}
```

Comment:
Adding some comment for this block of code will be helpful for better reading

Reply:
comments added.

Reply:
nit: fwiw comments should start with upper case and finalise with period. We should lint this.